### ISSUE TITLE
BitLocker automatic unlocking does not mount BTRFS correctly.

### ISSUE THREAD

**Issue by 3ri4nG0ld** [maharmstone/btrfs#549]
Currently you cannot create a BitLocker drive with BTRFS from the BitLocker graphical manager, but you can create a BitLocker encrypted NTFS drive and then format the NTFS partition to BTRFS (with ntfs2btrfs for example). 
This works perfectly for drives that you want to unlock with a password, but not for drives with automatic unlocking.

Steps to replicate:

1. Format our drive in NTFS.
2. Go to the BitLocker administrator and activate BitLocker on our drive.
3. When it asks us for the unlocking methods we add a password and select automatic unlocking (only available for fixed drives on the computer).
4. Once encrypted with BitLocker, we go to a Linux distribution, and unlock our BitLocker drive.
5. Format the NTFS partition to BTRFS.

If everything worked it should automatically unlock the disk and BTRFS would work inside BitLocker, but the disk shows an error and asks to format it.

But if we perform the same procedure but without activating the automatic unlocking, it can be unlocked without problems.

* Disk With automatic unlocking activated.
![image1](https://user-images.githubusercontent.com/24411137/212423927-5e28cb7b-9a23-4b9d-8ba6-7e36e5d7b881.png)

* Disk without automatic unlocking Activated (obviously the image is after unlocking the unit using the password)
![image2](https://user-images.githubusercontent.com/24411137/212424293-d62c5665-98d9-4a6a-84c7-823adce88e79.png)
(obviously the image is after unlocking the unit using the password)

**Additional information**
Try to enable automatic unlocking using the command `manage-bde -autounlock -enable E:`, but it drops the following error.
![image](https://user-images.githubusercontent.com/24411137/212424996-434a84de-c211-4bb2-91f7-25805204452c.png)

By the way, sorry for my English, I'm using a translator :P

**Comment by maharmstone** [comment]
Thank you, I'll see if I can reproduce it.

As an aside, I wouldn't recommend this "automatic unlocking" anyway - it looks to me very much like illusory security...

**Comment by 3ri4nG0ld** [comment]
> Thank you, I'll see if I can reproduce it.
> 
> As an aside, I wouldn't recommend this "automatic unlocking" anyway - it looks to me very much like illusory security...

Yeah, it's not the most secure, but it's the only thing in windows to encrypt a disk that is going to be used by the system. I currently have a disk set up with bitlocker and automatic unlocking for user folders, and I have it set up on my dualboot to automatically unlock it on my Arch and share user folders between both systems. By having the main disk of both systems encrypted I am less concerned about an encryption key being stored on them.

**Comment by mrdc** [comment]
> Thank you, I'll see if I can reproduce it.
> 
> As an aside, I wouldn't recommend this "automatic unlocking" anyway - it looks to me very much like illusory security...

Hi, was you able to reproduce it? I’m thinking to switch to Btrfs on some Windows PCs, but a few have Bitlocker drives with auto mounting.
Regarding security: if the system drive is also encrypted then auto mounted Bitlocker drives are quite secure - way better than have them in plaintext :D
